### ISSUE TITLE
Open MPI 4.1.1 from RHEL 9 repo, Slurm 23 mpirun Unable to alloc send buffer MR on mlx5_0: Cannot allocate memory; Unable to allocate UD send buffer pool

### ISSUE THREAD

**Issue by SomePersonSomeWhereInTheWorld** [open-mpi/ompi#12345]
Using OpenMPI 4.1.1 from a RHEL 9 repo, Slurm 23, and [this same job](https://usc-rc.github.io/tutorials/open-mpi), mpirun returns these errors/warnings.
```
2402-login-001:rank25.test_mpi: Unable to alloc send buffer MR on mlx5_0: Cannot allocate memory
2402-login-001:rank25.test_mpi: Unable to allocate UD send buffer pool
--------------------------------------------------------------------------
WARNING: There was an error initializing an OpenFabrics device.

  Local host:   ournode
  Local device: mlx5_0
--------------------------------------------------------------------------
```

```
Open MPI failed an OFI Libfabric library call (fi_endpoint).  This is highly
unusual; your job may behave unpredictably (and/or abort) after this.

  Local host: ournode
  Location: mtl_ofi_component.c:513
  Error: Invalid argument (22)
```

```
[1708109955.867962] [ournode:823516:0]          parser.c:1989 UCX  WARN  unused environment variable: UCX_IB_RCACHE_MAX_REGIONS
[1708109955.867962] [ournode:823516:0]          parser.c:1989 UCX  WARN  (set UCX_WARN_UNUSED_ENV_VARS=n to suppress this warning)
```
But finally starts:

```
Hello world from processor ournode, rank 60 out of 80 processors
Hello world from processor ournode, rank 74 out of 80 processors
```

Does OpenMPI need to be compiled with some other options?

**Comment by jsquyres** [comment]
Can you upgrade to the latest Open MPI (v4.1.x) and latest libfabric?

**Comment by SomePersonSomeWhereInTheWorld** [comment]
> Can you upgrade to the latest Open MPI (v4.1.x) and latest libfabric?

Well this is the latest from the RHEL 9 repo:
```
openmpi-devel-4.1.1-7.el9.x86_64
openmpi-4.1.1-7.el9.x86_64
```

**Comment by jsquyres** [comment]
Open MPI v4.1.1 was released in April of 2021; the latest Open MPI v4.1.x is v4.1.6 (Open MPI v5.0.2 also exists -- we haven't deprecated the v4.1.x series yet, though).

Looks like the latest Libfabric is 1.20.1 (https://github.com/ofiwg/libfabric/releases).

RHEL tends to move fairly slowly and not keep up with the latest releases.

**Comment by SomePersonSomeWhereInTheWorld** [comment]
> Open MPI v4.1.1 was released in April of 2021; the latest Open MPI v4.1.x is v4.1.6 (Open MPI v5.0.2 also exists -- we haven't deprecated the v4.1.x series yet, though).
> 
> Looks like the latest Libfabric is 1.20.1 (https://github.com/ofiwg/libfabric/releases).
> 
> RHEL tends to move fairly slowly and not keep up with the latest releases.

I took your word for it and compiled 5.0.2
```
mpirun --version
mpirun (Open MPI) 5.0.2
```
Not even one warning!
```
mpirun  ping_pong
Transfer size (B):          8, Transfer Time (s):     0.000000535, Bandwidth (GB/s):     0.013927360
Transfer size (B):         16, Transfer Time (s):     0.000000458, Bandwidth (GB/s):     0.032502642
Transfer size (B):         32, Transfer Time (s):     0.000000436, Bandwidth (GB/s):     0.068325743
Transfer size (B):         64, Transfer Time (s):     0.000000561, Bandwidth (GB/s):     0.106260398
Transfer size (B):        128, Transfer Time (s):     0.000000730, Bandwidth (GB/s):     0.163242256
Transfer size (B):        256, Transfer Time (s):     0.000000583, Bandwidth (GB/s):     0.408902154
Transfer size (B):        512, Transfer Time (s):     0.000000702, Bandwidth (GB/s):     0.679052076
Transfer size (B):       1024, Transfer Time (s):     0.000000915, Bandwidth (GB/s):     1.041709157
Transfer size (B):       2048, Transfer Time (s):     0.000001112, Bandwidth (GB/s):     1.715303278
Transfer size (B):       4096, Transfer Time (s):     0.000002479, Bandwidth (GB/s):     1.538817282
Transfer size (B):       8192, Transfer Time (s):     0.000003594, Bandwidth (GB/s):     2.122832002
Transfer size (B):      16384, Transfer Time (s):     0.000005450, Bandwidth (GB/s):     2.799829183
Transfer size (B):      32768, Transfer Time (s):     0.000009427, Bandwidth (GB/s):     3.237400043
Transfer size (B):      65536, Transfer Time (s):     0.000017176, Bandwidth (GB/s):     3.553447785
Transfer size (B):     131072, Transfer Time (s):     0.000019182, Bandwidth (GB/s):     6.363725165
Transfer size (B):     262144, Transfer Time (s):     0.000034139, Bandwidth (GB/s):     7.151313575
Transfer size (B):     524288, Transfer Time (s):     0.000062447, Bandwidth (GB/s):     7.819133127
Transfer size (B):    1048576, Transfer Time (s):     0.000119640, Bandwidth (GB/s):     8.162489938
Transfer size (B):    2097152, Transfer Time (s):     0.000221028, Bandwidth (GB/s):     8.836562914
Transfer size (B):    4194304, Transfer Time (s):     0.000418043, Bandwidth (GB/s):     9.344132890
Transfer size (B):    8388608, Transfer Time (s):     0.000819317, Bandwidth (GB/s):     9.535384325
Transfer size (B):   16777216, Transfer Time (s):     0.001544610, Bandwidth (GB/s):    10.115821397
Transfer size (B):   33554432, Transfer Time (s):     0.003069582, Bandwidth (GB/s):    10.180538742
Transfer size (B):   67108864, Transfer Time (s):     0.008313990, Bandwidth (GB/s):     7.517449178
Transfer size (B):  134217728, Transfer Time (s):     0.019888050, Bandwidth (GB/s):     6.285181419
Transfer size (B):  268435456, Transfer Time (s):     0.043864835, Bandwidth (GB/s):     5.699326170
Transfer size (B):  536870912, Transfer Time (s):     0.091848641, Bandwidth (GB/s):     5.443738700
Transfer size (B): 1073741824, Transfer Time (s):     0.187275044, Bandwidth (GB/s):     5.33973978
```


```
mpirun   ./test_mpi
Hello world from processor 2402-node004, rank 0 out of 12 processors
Hello world from processor 2402-node004, rank 6 out of 12 processors
Hello world from processor 2402-node004, rank 5 out of 12 processors
Hello world from processor 2402-node004, rank 7 out of 12 processors
Hello world from processor 2402-node004, rank 3 out of 12 processors
Hello world from processor 2402-node004, rank 2 out of 12 processors
Hello world from processor 2402-node004, rank 1 out of 12 processors
Hello world from processor 2402-node004, rank 8 out of 12 processors
Hello world from processor 2402-node004, rank 9 out of 12 processors
Hello world from processor 2402-node004, rank 4 out of 12 processors
Hello world from processor 2402-node005, rank 11 out of 12 processors
Hello world from processor 2402-node005, rank 10 out of 12 processor
```

Is this already listed as a bug on RHEL 9? Is there an option to `mpirun` or env variable that works around the 4.1.1 warnings? Thanks for the suggestion, perhaps there is a 4.1.1-x version that fixes this?

**Comment by jsquyres** [comment]
We don't make 4.1.1-x releases; we only have 4.1.x releases -- the latest is 4.1.6, released 30 Sep 2023.  

More specifically: we are not the RHEL 9 packagers of Open MPI; you'll need to contact them to ask.  4.1.6 should be backwards compatible with 4.1.1.  I.e., if the RHEL packagers want to upgrade to 4.1.6, it should be ok to do so (I would encourage that, rather than trying to back-port all the fixes from 4.1.6 back to 4.1.1).

Since you have a fix, I'll go ahead and close this issue.  Glad you got it working!
